api/src/test/java/org/openmrs/module/fgh/mpi/SnapshotEventProcessorTest.java

```diff
@@ -0,0 +1,143 @@
+	
+	@Before
+	public void setup() throws Exception {
+		PowerMockito.mockStatic(Context.class);
+		PowerMockito.mockStatic(MpiUtils.class);
+		PowerMockito.mockStatic(FhirUtils.class);
+		PowerMockito.mockStatic(MpiContext.class);
+		PowerMockito.mockStatic(KeyStore.class);
+		PowerMockito.mockStatic(KeyManagerFactory.class);
+		Whitebox.setInternalState(snapshotEventProcessor, MpiHttpClient.class, mockMpiHttpClient);
+		when(Context.getAdministrationService()).thenReturn(adminService);
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(AUTHENTICATION_TYPE.toString());
+		when(adminService.getGlobalProperty(GP_MPI_BASE_URL)).thenReturn(MPI_BASE_URL);
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM.toString());
+		when(adminService.getGlobalProperty(GP_MPI_APP_CONTENT_TYPE)).thenReturn(MPI_APP_CONTENT_TYPE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_FOCUS_REFERENCE)).thenReturn(MESSAGE_HEADER_REFERENCE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_ID)).thenReturn(SANTE_CLIENT_ID);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_SECRET)).thenReturn(SANTE_CLIENT_SECRET);
+		when(MpiUtils.getGlobalPropertyValue(GP_UUID_SYSTEM)).thenReturn(UUID_SYSTEM);
+		when(MpiContext.initIfNecessary()).thenReturn(mpiContext);
+		mpiContext.setAuthenticationType(AUTHENTICATION_TYPE);
+	}
+	
+	@Test
+	public void process_shouldIntegrateWithOpenCr() throws Exception {
+		AuthenticationType CERTIFICATE = AuthenticationType.CERTIFICATE;
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(CERTIFICATE.toString());
+		final String patientUuid = "patient-uuid-for-openCR";
+		Map res = new HashMap();
+		res.put("active", true);
+		when(mockMpiHttpClient.getPatient(patientUuid)).thenReturn(res);
+		Map prevState = singletonMap("uuid", patientUuid);
+		final Integer patientId = 1;
+		
+		when(MpiUtils.executeQuery(PERSON_QUERY.replace(ID_PLACEHOLDER, patientId.toString())))
+		        .thenReturn(asList(asList(null, null, null, null, patientUuid, null)));
+		
+		List<List<Object>> expectedPatient = new ArrayList<>();
+		expectedPatient.add(asList(patientUuid));
+		when(MpiUtils.executeQuery(PATIENT_QUERY.replace(ID_PLACEHOLDER, patientId.toString()))).thenReturn(expectedPatient);
+		Map<String, Object> patinetGeneratedPayload = new HashMap<>();
+		patinetGeneratedPayload.put("name", "mpi");
+		when(FhirUtils.buildPatient(patientId.toString(), false, expectedPatient.get(0), patinetGeneratedPayload))
+		        .thenReturn(patinetGeneratedPayload);
+		snapshotEventProcessor.process(new DatabaseEvent(null, "person", UPDATE, null, prevState, null));
+		Mockito.verify(mpiContext, Mockito.never()).initOauth();
+		
```

Comment:
Where are the actual checks here that ensure that the invoked logic was executed properly